include/zephyr/drivers/api.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file
+ * @brief Generic public API header file.
+ */
+
+/*
+ * Copyright (c) 2024 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_API_H_
```

Comment:
let's not introduce another header, device.h should be fine (maybe a subfamily in there). Also keep all under `DEVICE_` namespace, API is part of struct device after all.